src/pybind/mgr/dashboard/controllers/smb.py

```diff
@@ -0,0 +1,94 @@
+
+# -*- coding: utf-8 -*-
+
+import logging
+import json
+
+from typing import Dict, List, Union
+
+
+from dashboard.controllers._permissions import CreatePermission, DeletePermission
+from smb import results
+from smb.enums import AuthMode, SMBClustering, Intent
+
+from .. import mgr
+from ..security import Scope
+from . import APIDoc, APIRouter, ReadPermission, RESTController
+
+logger = logging.getLogger('controllers.smb')
+
+
+@APIRouter('/smb/cluster', Scope.SMB)
+@APIDoc("SMB Cluster Management API", "SMB")
+class SMBCluster(RESTController):
+    _resource: str = 'ceph.smb.cluster'
+
+    @ReadPermission
```

Comment:
let's add Endpoint documentation to every endpoint that we add.
```suggestion
    @ReadPermission
    @EndpointDoc("List all cluster resources")
```

Reply:
Sure, I'll update it

Reply:
Added docstrings, need to also add @endpointdoc